ultralytics/engine/model.py

```diff
@@ -443,13 +448,15 @@ def predict(
 
         if not self.predictor:
             self.predictor = predictor or self._smart_load("predictor")(overrides=args, _callbacks=self.callbacks)
+            self.end2end_arg(self.predictor)  # YOLOv10
             self.predictor.setup_model(model=self.model, verbose=is_cli)
         else:  # only update args if predictor is already setup
             self.predictor.args = get_cfg(self.predictor.args, args)
             if "project" in args or "name" in args:
                 self.predictor.save_dir = get_save_dir(self.predictor.args)
         if prompts and hasattr(self.predictor, "set_prompts"):  # for SAM-type models
             self.predictor.set_prompts(prompts)
+        self.end2end_arg(self.predictor)
```

Comment:
These are placed after the object (predictor, validator, trainer) initialization to avoid error with `default.yaml` config missing `end2end` argument.